### PR TITLE
Respect ENV in kernelspec

### LINKED ISSUE #14: Incorporate env key from kernelspecs
As mentioned in https://github.com/nteract/hydrogen/issues/404#issuecomment-239998118, the `env` key in the kernelspec is not being injected into the spawned kernel.

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #14.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -118,6 +118,7 @@ function launchSpec(kernelSpec, spawnOptions) {
     const connectionFile = c.connectionFile;
     const config = c.config;
     const argv = kernelSpec.argv.map(x => x === '{connection_file}' ? connectionFile : x);
+    spawnObjects['env'] = Object.assign({}, process.env, kernelSpec.env);
```
Comment (rgbkrk) [inline index.js]: Likely want `spawnOptions['env']` here.

Comment (captainsafia) [inline index.js]: DOH!

File: index.js [hunk 2]
```diff
@@ -141,7 +142,7 @@ function launchSpec(kernelSpec, spawnOptions) {
  * @return {string}       spawnResults.connectionFile  connection file path
  * @return {object}       spawnResults.config          connectionConfig
  */
-function launch(kernelName, spawnOptions, specs) {
+function launch(kernelName, spawnOptions = {}, specs) {
```
Comment (rgbkrk) [inline index.js]: We'll need babel config for the default assignment here. Fun fact though:

```
> Object.assign({}, {a: 2}, undefined)
{ a: 2 }
```

It's ok if `spawnOptions` is `undefined` for the `Object.assign`.

File: index.js (hunk 1 above)
Comment (captainsafia) [inline index.js]: Oh wow! The test failed on my local and I just blindly committed. I think it's time for my second lunch....

**Comment by lgeiger** [comment]
Awesome! I'll make a patch release to fix https://github.com/nteract/hydrogen/issues/404 soon.

**Pull request merged**